Adding new peering trobuleshooting page. (#1312)
hyperledger/besu-docs: docs/public-networks/how-to/troubleshoot/troubleshoot-peering.md

---
description: How to troubleshoot Peering
tags:
  - public networks
---

# Troubleshooting Peering

> Besu can have trouble peering for a number of reasons. It may have to do with your network router, machine environment, and node configuration. If you are having trouble, [configuring your ports](../connect/configure-ports.md) is a good place to start. Additionally, learning how to [manage peers](../connect/manage-peers.md) will help in maintaining healthy peer counts for your node. Below we have some frequently asked questions that may help you.

Reviewer: ```suggestion
A number of factors can affect the ability of your node to find and maintain peers. Your network router, machine environment, and node configuration are all important. If you are having peering issues, [configuring your ports](../connect/configure-ports.md) is a good place to start. Additionally, learning how to [manage peers](../connect/manage-peers.md) will help in maintaining healthy peer counts for your node. Below we have some frequently asked questions that may help you.
```

# Peering FAQ

**Q: Why can’t I find enough peers to sync?**

> A: Multiple possible reasons:

Reviewer: ```suggestion
A: 
```

- Poor hardware spec: not enough CPU, disk IOPS or bandwidth to handle all the peers
- Your ports are not open in your firewall and/or router
- Besu sending large numbers of DNS requests may cause issues, see [issue #4375.](https://github.com/hyperledger/besu/issues/4375)
- It is normal for peers to connect and disconnect, especially soon after you start your node.
- Try setting [`p2p-host`](../../reference/cli/options.md#p2p-host) to your external IP address to allow inbound connections

> - In proof of stake, we can set [`fast-sync-min-peers`](../../reference/cli/options.md#fast-sync-min-peers) to less than 5, for example 3.

Reviewer: I think we changed the default to 3 or 1 so this one may be no longer relevant

- Restarting may get new peers, however, it can take a while to build up again.
- `-Xdns-enabled=true`
- `-Xp2p-peer-lower-bound` can be set to a minimum number of peers
- Deleting the node key (which is autogenerated in your data directory) may help with finding new peers. Two potential reasons for this:
    1. Your node (identified by the address associated with this key) has been put onto other peers' bad peer lists for some reason.
    2. Peer discovery is actually influenced by the value of the node key, it has to do with the node 'distance' in the [discovery algorithm.](https://github.com/ethereum/devp2p/wiki/Discovery-Overview#kademlia)

The Prysm team has good documentation on [EL + CL peering](https://docs.prylabs.network/docs/prysm-usage/p2p-host-ip) that may help.

**Q: How do I open/forward my ports?**

> A: If you’re behind NAT, you probably need to port forward in your router. You may also need to configure your firewall. Forward and open `30303` (if using default p2p port) for both UDP and TCP. If your router supports UPNP, you can setting the [`--nat-method`](../../reference/cli/options.md#nat-method) to `UPNPP2PONLY`. You can learn more about [NAT methods here.](../connect/specify-nat.md)

Reviewer: ```suggestion
A: If you’re behind NAT, you probably need to set up port forwarding in your router. You may also need to configure your firewall. Forward and open `30303` (if using default p2p port) for both UDP and TCP. If your router supports UPNP, you can setting the [`--nat-method`](../../reference/cli/options.md#nat-method) to `UPNPP2PONLY`. You can learn more about [NAT methods here.](../connect/specify-nat.md)
```

**Q: How do I test that my ports are open?**

A: You can test with [Open Port Check Tool.](https://www.yougetsignal.com/tools/open-ports/)

**Q: What is the ideal number of peers for Besu?**

A: Default max is 25. Increasing the number of peers increases the bandwidth, CPU and disk access Besu uses to respond to peers. A poor hardware spec may be the reason for low peer numbers. You will experience diminishing returns with a larger number of peers (> 100).

**Q: What is the benefit of increasing the number of peers?**

A: Increasing the number of max peers will not speed up Besu sync, as the bottleneck during sync is Disk IO and CPU.

> Bear in mind that Besu's peers are only used for initial sync and transaction gossip, neither of which affects attestation performance. It's the beacon node connectivity that controls how quickly your receive blocks and how attestations are published. Increasing Besu's peer count won't help with attestations and will increase the load on your node which may make things worse.

Reviewer: ```suggestion
Bear in mind that Besu's peers are only used for initial sync and transaction gossip, neither of which affects attestation performance. It's the beacon node connectivity that controls how quickly you receive blocks and how attestations are published. Increasing Besu's peer count won't help with attestations and will increase the load on your node which may make things worse.
```

## Metrics

> Make sure you are capturing metrics to gain insights into peer values over time.

Reviewer: ```suggestion
Capturing metrics will help to gain insights into peering behavior over time.
```

> To enable

Reviewer: are these instructions already specified elsewhere in the docs? would it be better to link there
